src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -67,7 +70,10 @@ class CodyToolWindowContent(private val project: Project) {
 
   fun removeAllChatSessions() {
     AgentChatSessionService.getInstance(project).removeAllSessions()
-    switchToChatSession(AgentChatSession.createNew(project))
+    UIUtil.invokeLaterIfNeeded {
+      chatContainerPanel.removeAll()
+      chatContainerPanel.add(JBLabel(AnimatedIcon.Default()))
```

Comment:
I do not like that loading screen on account change. Why do we need it?  We do have a default model? There is no loading screen on startup. If we need it to wait for the models list why don't we just use the default (since we define it) OR display it later like for startup?

Reply:
When we are switching an account we need to load LLM dropdown before user is able to send a message. If i use default model (hardcoded one)  it might not be in line with default model for enterprise users...

Reply:
That is not exactly true. If you use default model (just by not specifying it) it will always pick correct default.